python/ipywidgets/ipywidgets/comm.py

```diff
@@ -0,0 +1,33 @@
+# compatibility shim for ipykernel < 6.18
+import sys
+from IPython import get_ipython
+import comm
+
+
+def requires_ipykernel_shim():
+    if "ipykernel" in sys.modules:
+        import ipykernel
+
+        version = ipykernel.version_info
+        return version < (6, 18)
+    else:
+        return False
+
+
+def get_comm_manager():
+    if requires_ipykernel_shim():
+        ip = get_ipython()
+
+        if ip is not None and getattr(ip, "kernel", None) is not None:
+            return get_ipython().kernel.comm_manager
+    else:
+        return comm.get_comm_manager()
+
+
+def create_comm(*args, **kwargs):
+    if requires_ipykernel_shim():
+        from ipykernel.comm import Comm
+
+        return Comm(*args, **kwargs)
```

Comment:
I'd really like to get rid of this some day.

That way we can remove the nasty patches of ipykernel.comm in xeus-python.

Reply:
only for old ipykernels right? Do you think that will still happen?

Reply:
I'd like this piece of code to basically stay around forever, so we better get it right! :)
We can also skip this if we can detect xeus-python, or some other condition to make this more robust? Do you make use of a 'fake' ipykernel module in xeus-python?

Reply:
Only for old ipykernel yes.

> I'd like this piece of code to basically stay around forever

Let's talk again about it in ten years? 😄 

> Do you make use of a 'fake' ipykernel module in xeus-python?

Yeah we mock the ipykernel module https://github.com/jupyter-xeus/xeus-python/blob/main/src/xinterpreter.cpp#L77

Reply:
Happy to discuss this in ten years from now :)